src/ansys/speos/core/generic/constants.py

```diff
@@ -38,3 +42,347 @@
+
+    wavelength_range: WavelengthsRangeParameters = field(default_factory=WavelengthsRangeParameters)
+    """Wavelength range of the Sensor."""
+
+
+@dataclass
+class SpectralParameters:
+    """Colorimetric settings of the Sensor."""
+
+    wavelength_range: WavelengthsRangeParameters = field(default_factory=WavelengthsRangeParameters)
+    """Wavelength range of the Sensor."""
+
+
+class IntegrationTypes(str, Enum):
+    """Integration types without parameters."""
+
+    planar = "planar"
+    radial = "radial"
+    hemispherical = "hemispherical"
+    cylindrical = "cylindrical"
+    semi_cylindrical = "semi_cylindrical"
+
+
+class RayfileTypes(str, Enum):
+    """Rayfile types without parameters."""
+
+    none = "none"
+    classic = "classic"
+    polarization = "polarization"
+    tm25 = "tm25"
+    tm25_no_polarization = "tm25_no_polarization"
+
+
+class SensorTypes(str, Enum):
+    """Sensor types without parameters."""
+
+    photometric = "photometric"
+    radiometric = "radiometric"
+
+
+@dataclass
+class MeasuresParameters:
+    """Measurements for 3d Irradiance Sensor."""
+
+    reflection: bool = True
+    """Reflection measure activation state."""
+    transmission: bool = True
+    """Transmission measure activation state."""
+    absorption: bool = True
+    """Ansorption measure activation state."""
+
+
+@dataclass
+class IrradianceSensorParameters:
+    """Irradiance Sensor Parameters."""
+
+    dimensions: DimensionsParameters = field(default_factory=DimensionsParameters)
+    """Dimensions of the sensor."""
+    axis_system: list[float] = field(default_factory=lambda: ORIGIN)
+    """Position of the sensor."""
+    sensor_type: Union[
+        SensorTypes.photometric, ColorimetricParameters, SpectralParameters, SensorTypes.radiometric
+    ] = SensorTypes.photometric
+    """Type of the sensor."""
+    integration_type: Union[IntegrationTypes] = IntegrationTypes.planar
+    """Integration type of the sensor."""
+    integration_direction: Union[None, list[float]] = None
+    """Integration direction of the sensor."""
+    rayfile_type: Union[RayfileTypes] = RayfileTypes.none
+    """Type of rayfile stored by the sensor."""
+    layer_type: Union[
+        LayerTypes,
+        LayerByFaceParameters,
+        LayerBySequenceParameters,
+        LayerByIncidenceAngleParameters,
+    ] = LayerTypes.none
+    """Type of layer separation used by the sensor."""
+    outpath_face_geometry: list = None
+    """Outpath face used by the sensor"""
+
+
+@dataclass
+class RadianceSensorParameters:
+    """Radiance Sensor Constants."""
+
+    dimensions: DimensionsParameters = field(default_factory=DimensionsParameters)
+    """Dimensions of the sensor."""
+    axis_system: list[float] = field(default_factory=lambda: ORIGIN)
+    """Position of the sensor."""
+    sensor_type: Union[
+        SensorTypes.photometric, ColorimetricParameters, SpectralParameters, SensorTypes.radiometric
+    ] = SensorTypes.photometric
```

Comment:
same. I dont know if there is a better and cleaner way to do that. best case would be SensorTypes which contains all the types. Need to ask Sebastien